src/Symfony/Contracts/Service/ServiceSubscriberInterface.php

```diff
@@ -47,7 +47,7 @@ interface ServiceSubscriberInterface
      *  * ['?Psr\Log\LoggerInterface'] is a shortcut for
      *  * ['Psr\Log\LoggerInterface' => '?Psr\Log\LoggerInterface']
      *
-     * @return array The required service types, optionally keyed by service names
+     * @return string[] The required service types, optionally keyed by service names
```

Comment:
Is it really better? Reading the examples in the phpdocs, it does not seem right to me. Note that I'm not a phpdoc expert.

Reply:
this is right. This is indeed an array with string values (and any valid array keys, i.e. integers or strings)

Reply:
The more precise annotation would be this:

```suggestion
     * @return array<int|string, string> The required service types, optionally keyed by service names
```

I wasn't sure if we actively use this generic-style array type notation in this codebase, so I picked a more convervative version.

Reply:
I use the same in my projects @derrabus

Reply:
If I'm not wrong, the list of allowed combinations are:
```
return [
  'event' => 'method',
  'event' => ['method', 10],
  'event' => [['method', 10], ['method', 10]]
];
```

In that case, the signature should be `array<string, string|array{string, int}|list<array{string, int}>>`...

Reply:
Why not just keep `array`?

Reply:
Its just an assumption here, but many people use static code analysers and need to repeat to add `array<int|string, string>` over and over again. It would not hurt us but adds benefit to the end user, because its more helpful, the more strict it is. Am I right @derrabus or is there any other intention?

Reply:
A more precise return type would help a static code analysis determine if my userland implementation of `getSubscribedServices()` is returning the right data structure.

Here is an example of PHPStan with a fairly low rule level: https://phpstan.org/r/b8854d54-ba65-4427-9a62-453376e2dc44